code/game/machinery/doors/airlock.dm

```diff
@@ -1131,8 +1131,8 @@
 
 /obj/machinery/door/airlock/try_to_weld(obj/item/weapon/weldingtool/W, mob/user)
 	if(!operating && density)
-		if(W.remove_fuel(0,user))
-			if(user.a_intent != INTENT_HELP)
+		if(user.a_intent != INTENT_HELP)
+			if(W.remove_fuel(0,user))
```

Comment:
That's already what happens basically, all `remove_fuel(0, ...)` really does is blind the person if necessary